File: my-code/1996/main.py
# -*- coding: UTF-8 -*-
# @Time     : 2022/01/28 21:37
# @Author   : Ranshi
# @File     : main.py
# @Doc      : 1996. 游戏中弱角色的数量
class Solution:
    def numberOfWeakCharacters(self, properties: list[list[int]]) -> int:
        properties.sort(key=lambda x: (-x[0], x[1]))
        ans = 0
        maxDef = 0
        for _, def_ in properties:
            if maxDef > def_:
                ans += 1
            else:
                maxDef = max(maxDef, def_)
        return ans


if __name__ == "__main__":
    print(
        Solution().numberOfWeakCharacters(
            properties=[[7, 9], [10, 7], [6, 9], [10, 4], [7, 5], [7, 10]]
        )
    )

File: my-code/1996/main.go
package main

import (
	"fmt"
	"sort"
)

// @Time     : 2022/01/28 22:06
// @Author   : Ranshi
// @File     : main.go
// @Doc      : 1996. 游戏中弱角色的数量
func numberOfWeakCharacters(properties [][]int) (ans int) {
	sort.Slice(properties, func(i, j int) bool {
		p, q := properties[i], properties[j]
		return p[0] > q[0] || p[0] == q[0] && p[1] < q[1]
	})
	maxDef := 0
	for _, p := range properties {
		if p[1] < maxDef {
			ans++
		} else {
			maxDef = p[1]
		}
	}
	return
}

func main() {
	fmt.Println(numberOfWeakCharacters([][]int{
		{1, 5},
		{10, 4},
		{4, 3},
	}))
}
